worlds/stardew_valley/__init__.py

```diff
@@ -439,9 +433,18 @@ def collect(self, state: CollectionState, item: StardewItem) -> bool:
         if not change:
             return False
 
+        player_state = state.prog_items[self.player]
+
+        received_progression_count = player_state[Event.received_progression_item]
+        received_progression_count += 1
+        if self.total_progression_items:
+            # We can't update the percentage if we don't know the total progression items, can't divide by 0.
```

Comment:
It might be good to add a comment saying _when_ this is even possible. I'm guessing it's during tests, but it definitely seems like something that should be impossible during regular generation

Reply:
Since I _think_ this is always true during regular generations, it does kind of suck that you're having to check an extra `if` every single `collect/remove`. Are there problems with either making `total_progression_items` 1 by default of making the zero division just return 0 (or 100) instead?

Reply:
It can happen during the create item step, when precollecting seasons for instance.

Reply:
Then I'll go to my other point. Is there any way for this to actually matter at that time? `received_progression_percent` is always set to a value, never incremented/decremented, so just setting it to something like 1 by default or making this return 100 should be fine?

Reply:
I think you could even just make precollecting increment the value directly instead?

Reply:
I tried setting the `total_progression_items` to 1 or 100, and that ended up with generation errors. The collection state is used directly without any new items getting collected after the `total_progression_items` is updated to its real value, so the `received_progression_percent` ended up with a bad value during the first steps of the fill. However, I found out that setting the default value to `-1` does work, since it makes the `received_progression_percent` negative, it always acts as if there were no progression items collected when starting the fill. 

In any case, the impact on the generation time seems minimal, if any. It might be 1% faster, but that is inside the standard deviation, so that might just be a fluke.

![image](https://github.com/user-attachments/assets/cc2e1282-9ce5-4521-899d-fd4422dfb17c)

Reply:
I added the comment, but I don't think adding the default value is a good idea in the end. I feel like it's easier to understand that the progression item count is unknown until all items are created than using a default value to avoid a division by 0.